### PR TITLE
BPM-33 Add Travis Integration

### PR DESCRIPTION
**Pull request by datmau**
-Manage Basic Endpoints
-Add Travis and CodeCov Integration.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+cache:
+  directories:
+    - $HOME/.gradle/caches/
+    - $HOME/.gradle/wrapper/
+after_success:
+  - bash <(curl -s https://codecov.io/bash)
+  - pip install --user awscli
+  - eval $(aws ecr get-login --no-include-email --region us-east-1)
```
Comment (enriquezrene) [inline .travis.yml:15]: We don't need the last two lines.

Comment (enriquezrene) [reply .travis.yml:15]: Those lines are being used to push a docker image to ecr which this project does not have yet.

File: build.gradle [hunk 2]
```diff
@@ -1,6 +1,6 @@
 buildscript {
     ext {
-        springBootVersion = '2.1.0.RELEASE'
+        springBootVersion = '2.0.1.RELEASE'
     }
```
Comment (enriquezrene) [inline build.gradle:4]: Why we need 2.01.RELEASE instead of 2.1.0.RELEASE?

File: build.gradle [hunk 3]
```diff
@@ -33,19 +37,45 @@ ext {
 }
 
 dependencies {
-    testImplementation 'org.mockito:mockito-core:2.7.22'
     compile 'com.github.derjust:spring-data-dynamodb:5.0.3'
     compile('org.springframework.boot:spring-boot-starter-web')
     runtime('org.springframework.boot:spring-boot-devtools')
     compile 'io.springfox:springfox-swagger2:2.7.0'
     compile 'io.springfox:springfox-swagger-ui:2.7.0'
```
Comment (enriquezrene) [inline build.gradle:44]: You might want to update the swagger dependency version to: 2.9.2

File: src/main/java/com/ioet/bpm/bpmorganizationsapi/boundaries/OrganizationController.java [hunk 4]
```diff
@@ -8,20 +8,22 @@
 import io.swagger.annotations.ApiResponse;
```
Comment (enriquezrene) [inline src/main/java/com/ioet/bpm/bpmorganizationsapi/boundaries/OrganizationController.java:8]: The package name is really weird. I suggest com.ioet.bpm.organizations.boundaries instead.

File: src/main/java/com/ioet/bpm/bpmorganizationsapi/boundaries/OrganizationController.java [hunk 5]
```diff
@@ -30,11 +32,69 @@
...
     }
 
+    @ApiOperation(value = "Return one organization by id", response = Organization.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Organization returned")
+    })
+
+    @GetMapping(path = "/{id}" , produces = "application/json")
+
+    public ResponseEntity<Organization> getOrganization(@PathVariable(value = "id") String organizationId){
```
Comment (enriquezrene) [inline src/main/java/com/ioet/bpm/bpmorganizationsapi/boundaries/OrganizationController.java:47]: You can remove the break lines.

File: src/main/java/com/ioet/bpm/bpmorganizationsapi/boundaries/OrganizationController.java [hunk 6]
```diff
@@ -30,11 +32,69 @@
...
+        Optional<Organization> personOptional = organizationRepository.findAllByUid(organizationId);
+        return personOptional.map(
+                person -> new ResponseEntity<>(person, HttpStatus.OK))
+                .orElseGet(() -> new ResponseEntity<>(HttpStatus.NOT_FOUND));
```
Comment (enriquezrene) [inline src/main/java/com/ioet/bpm/bpmorganizationsapi/boundaries/OrganizationController.java:51]: person?

File: src/main/java/com/ioet/bpm/bpmorganizationsapi/boundaries/OrganizationController.java [hunk 7]
```diff
@@ -30,11 +32,69 @@
...
 
+
+    @ApiOperation(value= "Delete an organization")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Organization successfully deleted"),
+            @ApiResponse(code = 404, message = "Organization to delete not found")
+    })
+
+    @DeleteMapping(path="/{id}", produces = "application/json")
+    public ResponseEntity<Organization> deleteOrganization(@PathVariable(value= "id") String organizationId ){
```
Comment (enriquezrene) [inline src/main/java/com/ioet/bpm/bpmorganizationsapi/boundaries/OrganizationController.java:74]: Remove break lines.

File: src/main/java/com/ioet/bpm/bpmorganizationsapi/boundaries/OrganizationController.java [hunk 8]
```diff
@@ -30,11 +32,69 @@
...
+    }
+
+    @ApiOperation(value = "Update an Organization")
+    @ApiResponses(value={
+            @ApiResponse(code = 200, message = "Organization sucessfully updated"),
+            @ApiResponse(code = 404, message = "Organization to delete not found")
+    })
+
+
+    @PutMapping(path="/{id}", produces = "application/json")
```
Comment (enriquezrene) [inline src/main/java/com/ioet/bpm/bpmorganizationsapi/boundaries/OrganizationController.java:90]: Remove break lines.

File: src/main/java/com/ioet/bpm/bpmorganizationsapi/config/DynamoDBConfig.java [hunk 9]
```diff
@@ -0,0 +1,64 @@
+package com.ioet.bpm.bpmorganizationsapi.config;
```
Comment (enriquezrene) [inline src/main/java/com/ioet/bpm/bpmorganizationsapi/config/DynamoDBConfig.java:1]: Change the package name, com.ioet.organizations.config

File: src/main/java/com/ioet/bpm/bpmorganizationsapi/entities/Organization.java [hunk 10]
```diff
@@ -23,20 +23,4 @@
     @DynamoDBAttribute
     private String organizationName;
 
-
```
Comment (enriquezrene) [inline src/main/java/com/ioet/bpm/bpmorganizationsapi/entities/Organization.java:26]: I think this class should have two fields: id and name

File: src/main/java/com/ioet/bpm/bpmorganizationsapi/repositories/OrganizationRepository.java [hunk 11]
```diff
@@ -7,13 +7,16 @@
...
 
 
 @EnableScan
 public interface OrganizationRepository extends CrudRepository <Organization, String> {
 
-    public List<Organization> findAll();
+    public List<Organization> findAllByUid();
+
+    public  Optional<Organization> findAllByUid(String uid);
+
```
Comment (enriquezrene) [inline src/main/java/com/ioet/bpm/bpmorganizationsapi/repositories/OrganizationRepository.java:20]: What the heck is doing this service?

File: src/test/java/com/ioet/bpm/bpmorganizationsapi/BpmorganizationsapiApplicationTests.java [hunk 12]
```diff
@@ -22,6 +24,9 @@
     @Mock
     OrganizationRepository repository;
 
+    @Mock
+    OrganizationController organizationController;
+
     @Test
     public void contextLoads() {
     }
```
Comment (enriquezrene) [inline src/test/java/com/ioet/bpm/bpmorganizationsapi/BpmorganizationsapiApplicationTests.java:32]: What's is doing this method here?

File: src/test/java/com/ioet/bpm/bpmorganizationsapi/BpmorganizationsapiApplicationTests.java [hunk 13]
```diff
@@ -81,8 +86,29 @@ public void listAllOrganizations(){
...
+        Organization organizationToCreate = mock(Organization.class);
+        Organization organizationCreated = mock(Organization.class);
+        Optional<Organization> organization = Optional.of(Mockito.mock(Organization.class));
+
 
+        when(organizationCreated.getUid()).thenReturn("1");
+        assertEquals("1",organizationCreated.getUid());
+
+        verify(organizationCreated, times(1)).getUid();
     }
```
Comment (enriquezrene) [inline src/test/java/com/ioet/bpm/bpmorganizationsapi/BpmorganizationsapiApplicationTests.java:100]: These tests don't make sense at all.

**Review by enriquezrene — CHANGES_REQUESTED** [review]
Please address the comments.

File: build.gradle (hunk 2 above)
Comment (datmau) [reply build.gradle:4]: Changing to that version solved all the problems I had in the build.

File: src/test/java/com/ioet/bpm/bpmorganizationsapi/BpmorganizationsapiApplicationTests.java (hunk 12 above)
Comment (datmau) [reply src/test/java/com/ioet/bpm/bpmorganizationsapi/BpmorganizationsapiApplicationTests.java:32]: This is generated by the IDE. I did forgot to delete it.

File: src/main/java/com/ioet/bpm/bpmorganizationsapi/repositories/OrganizationRepository.java (hunk 11 above)
Comment (datmau) [reply src/main/java/com/ioet/bpm/bpmorganizationsapi/repositories/OrganizationRepository.java:20]: I will delete it. Actually I dont know why I wrote that.

File: src/main/java/com/ioet/bpm/bpmorganizationsapi/boundaries/OrganizationController.java (hunk 4 above)
Comment (datmau) [reply src/main/java/com/ioet/bpm/bpmorganizationsapi/boundaries/OrganizationController.java:8]: I will change that.

File: src/test/java/com/ioet/bpm/bpmorganizationsapi/BpmorganizationsapiApplicationTests.java (hunk 13 above)
Comment (datmau) [reply src/test/java/com/ioet/bpm/bpmorganizationsapi/BpmorganizationsapiApplicationTests.java:100]: @enriquezrene all the test or only this one?

Comment (enriquezrene) [reply src/test/java/com/ioet/bpm/bpmorganizationsapi/BpmorganizationsapiApplicationTests.java:100]: none

File: .travis.yml [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+before_cache:
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+cache:
+  directories:
+    - $HOME/.gradle/caches/
+    - $HOME/.gradle/wrapper/
+after_success:
+  - bash <(curl -s https://codecov.io/bash)
+  - pip install --user awscli
```
Comment (enriquezrene) [inline .travis.yml:14]: You don't need AWS here.

File: build.gradle [hunk 15]
```diff
@@ -33,19 +37,45 @@ ext {
...
         mavenBom "org.springframework.cloud:spring-cloud-dependencies:${springCloudVersion}"
     }
 }
+
+test {
+    dependsOn 'cleanTest'
+    jacoco {
+        excludes += [
+                '**/config/**',
+                '**/BpmorganizationsapiApplication.class'
```
Comment (enriquezrene) [inline build.gradle:61]: This class name was changed

File: build.gradle [hunk 16]
```diff
@@ -33,19 +37,45 @@ ext {
...
+
+jacocoTestReport {
+    reports {
+        xml.enabled = true
+        html.enabled = true
+    }
+    jacocoTestReport {
+        afterEvaluate {
+            classDirectories = files(classDirectories.files.collect {
+                fileTree(dir: it, exclude: ['**/config/**', '**/BpmorganizationsapiApplication.class'])
```
Comment (enriquezrene) [inline build.gradle:74]: This classname was changed

**Review by enriquezrene — CHANGES_REQUESTED** [review]

**Pull request closed without merging**